Sources/MongoSwift/MongoCursor.swift

```diff
@@ -187,47 +181,59 @@ public class MongoCursor<T: Codable>: Sequence, IteratorProtocol {
     }
 
     /**
-     * Returns the next `Document` in this cursor or `nil`, or throws an error if one occurs -- compared to `next()`,
-     * which returns `nil` and requires manually checking for an error afterward.
-     * - Returns: the next `Document` in this cursor, or `nil` if at the end of the cursor
+     * Returns an array of type  `T` from this cursor.
+     * - Returns: an array of type `T`
      * - Throws:
-     *   - `ServerError.commandError` if an error occurs on the server while iterating the cursor.
+     *   - `ServerError.commandError` if an error occurs while fetching more results from the server.
      *   - `UserError.logicError` if this function is called after the cursor has died.
      *   - `UserError.logicError` if this function is called and the session associated with this cursor is inactive.
      *   - `DecodingError` if an error occurs decoding the server's response.
      */
-    public func nextOrError() throws -> T? {
-        if let next = self.next() {
-            return next
-        }
-        if let error = self.error {
-            throw error
+    public func all() throws -> [T] {
+        return try self.map {
+            switch $0 {
+            case let .success(t):
+                return t
+            case let .failure(error):
+                throw error
+            }
         }
-        return nil
     }
 
-    /// Returns the next `Document` in this cursor, or `nil`. After this function returns `nil`, the caller should use
-    /// the `.error` property to check for errors. For tailable cursors, users should also check `isAlive` after this
-    /// method returns `nil`, to determine if the cursor has the potential to return any more data in the future.
-    public func next() -> T? {
+    /**
+     * Returns a `Result` containing the next `T` in this cursor,  or an error if one occurred.
+     * Returns `nil` if the cursor is exhausted. For tailable cursors,
+     * users should also check `isAlive` after this method returns `nil`,
+     * to determine if the cursor has the potential to return any more data in the future.
+     * - Returns: `nil` if the end of the cursor has been reached, or a `Result`. On success, the
+     *   `Result` contains the next `T`, and on failure contains:
+     *   - `ServerError.commandError` if an error occurs while fetching more results from the server.
```

Comment:
After SWIFT-502, these enum based errors were replaced with structs, so the prefixes can be dropped.
e.g. "ServerError.commandError" becomes just "CommandError"

Reply:
good point. @ali-mir, were these a copy-paste from somewhere else in the driver we didn't update as part of SWIFT-502?

Reply:
good catch! @kmahar I copy pasted it from the `nextOrError()` docstring before this PR was merged. Updating the docstrings now.